### ISSUE TITLE
Bad import in lib/document.py

### ISSUE THREAD

**Issue by 0ion9** [mypaint/mypaint#419]
I'm not sure how this ever worked, but, when I try to run mypaint (git 0bc26c6), it promptly crashes:

Traceback (most recent call last):
  File "/usr/local/bin/mypaint", line 462, in <module>
    main.main(datapath, iconspath, old_confpath, version=version)
  File "/usr/local/share/mypaint/gui/main.py", line 77, in main
    from gui import application
  File "/usr/local/share/mypaint/gui/application.py", line 18, in <module>
    import autorecover
  File "/usr/local/share/mypaint/gui/autorecover.py", line 22, in <module>
    import lib.document
  File "/usr/local/share/mypaint/lib/document.py", line 20, in <module>
    import xml.etree.ElementTree as ET
ImportError: No module named etree.ElementTree

I have run 'python2.7' and tested that 'import xml.etree.ElementTree as ET' works okay.

AFAICS this is because you are not actually importing from the global xml module, but from lib/xml.py, which indeed doesn't have a submodule named etree.

Another way to demonstrate the bug is to `cd lib/` and launch python2.7, then enter `import xml` and watch the traceback.

There are two options: Rename lib/xml.py to eg. lib/mxml.py so it doesn't stomp on the global module namespace, or `from __future__ import absolute_import` (Using properly distinguished local vs global imports might be a good idea as a transitional step anyway, since it is mandatory in Py3, and just plain clearer.)

**Comment by 0ion9** [comment]
I have managed to get MyPaint to launch and work okay by modifying document.py like this:,
```
from __future__ import absolute_import
[...]
from . import helpers
from . import fileutils
from . import tiledsurface
from . import pixbufsurface
from . import mypaintlib
from . import command
from . import stroke
from . import layer
from . import brush
from .observable import event
import lib.pixbuf
from lib.errors import FileHandlingError
from lib.errors import AllocationError
import lib.idletask
from lib.gettext import C_
```

However, note that it brings up an error upon exit:

```
Traceback (most recent call last):
  File "/usr/local/share/mypaint/gui/drawwindow.py", line 681, in quit_cb
    self.app.profiler.cleanup()
AttributeError: 'Application' object has no attribute 'profiler'
```

**Comment by achadwick** [comment]
Looks like your libs in `/usr/local` are an old version. Check out <https://github.com/mypaint/mypaint/blob/0bc26c65365a1fe0846bd2cb53df0e8eaf2e9326/lib/document.py> and note blob id - same as yours, 0bc26c65365a1fe0846bd2cb53df0e8eaf2e9326. It already does the absolute import thing, and the `xml.etree` import is on line 22, not line 20.

Can you uninstall and reinstall from master please, and see if the error disappears?

I consider the original reported problem fixed with 5bb770b45191965ee0bc246d4503233bb28b7e0e (1 Jul), but there may be something else going on?

I consider the second reported problem fixed with bae0db944dc1dd2afd7a5026d36cd395bd4f372e (10 Aug).

Can you check that both of these commits are fully on your branch?

**Comment by 0ion9** [comment]
Ugh, synchronization glitches. Yeah, that's the cause. Wiping /usr/local/share/mypaint and reinstalling fixed it completely. Closing

(I've had this kind of problem before. It makes me wonder about how reliable SCons' tracking of the uptodateness of files is.)
